plugins/uefi/fu-plugin-uefi.c

```diff
@@ -439,6 +440,10 @@ fu_plugin_uefi_load_config (FuPlugin *plugin, FuDevice *device)
 	fu_device_set_metadata_boolean (device,
 					"RequireShimForSecureBoot",
 					!disable_shim);
+
+	/* only used to debug the EFI binary itself */
+	efi_debug_log = fu_plugin_get_config_value_boolean (plugin, "EfiDebugLog");
```

Comment:
Hmm rather than a new variable, why not just hook off `FWUPD_UEFI_VERBOSE` key that is already used to do verbose stuff and has the extra trigger from `fwupdtool --plugin-verbose=uefi`

Reply:
My first thought was that we should retain compatibility with the `fwupdate -D` and `-d` mode where QA users can just "_turn on_" extra debugging on a machine. I guess the alternate is we just explain to the user that the `-d` mode has been removed and that they should do fwupdmgr --enable-extra-debugging or something -- that would be a FuInstallFlag I guess, although I don't know if that's over-engineering the whole thing. Nowadays the EFI debugging is used less and less now we know where the bodies are hidden.

Reply:
I guess your point being we need the debugging to be available from `fwupdgmr`?  I'd say that's overengineering.  If someone has a problem (which it's been so long since we needed it), send them over to `fwupdate` to install it or or `fwupdtool --plugin-verbose=uefi`.

Reply:
So maybe we simplify it further, and just unconditionally delete FWUPDATE_VERBOSE and FWUPDATE_DEBUG_LOG when deploying the update using fwupdtool/fwupd and leave it enabled and readable by fwupdate -- if you want to debug the efi binary you then have to use fwupdate and squirt the .cap file rather than using all the other layers.

Reply:
Yes, that's great idea.

Reply:
Something like this^^?